doc: improve worker_threads documentation (#26110)
nodejs/node: doc/api/worker_threads.md

> Stability: 1 - Experimental

The `worker_threads` module enables the use of threads that execute JS code

> in parallel. To access it:

Reviewer: Maybe something more like "that allow parallel execution contexts for JS code"?

Author: @Fishrock123 I’m not sure … “execution context” sounds like a very generic thing?

> in parallel, with message channels between them. To access it:

Reviewer 2: Nit: Maybe drop the message channels stuff to focus on The One Thing you want someone to know about the module?:

```markdown
The `worker_threads` module enables the use of threads that execute code in parallel.
```

Author: done!

> An arbitrary JavaScript value that contains a clone of the data passed

Reviewer 3: It might be good to include more of a description of what "clone" here means in general. I know there are a few bits and pieces of an explanation through the doc, but coalescing those into a single section with some information on what types of values *cannot* be cloned, would be good.

Author: @jasnell How does 919fd07151495f55c60c59b79862962d459e1665 sound to you?

An arbitrary JavaScript value that contains a clone of the data passed
to this thread’s `Worker` constructor.

The data is cloned as if using [`postMessage()`][`port.postMessage()`],

> using the [HTML structured clone algorithm][].

Reviewer 4: Deduplicate "using... using..."?

This method can be called when no further communication will happen over this
`MessagePort`.

> The [`'close'` event][] will be emitted on both `MessagePort` instances that

Reviewer 2: ```suggestion
Both `MessagePort` instances that are part of the channel will emit a [`'close'` event][].
```

Author: I have to admit that active voice sounds a bit weird to me when referring to events :)

Sends a JavaScript value to the receiving side of this channel.
`value` will be transferred in a way which is compatible with
the [HTML structured clone algorithm][].

In particular, the significant differences to `JSON` are:
- `value` may contain circular references.
- `value` may contain instances of builtin JS types such as `RegExp`s,
  `BigInt`s, `Map`s, `Set`s, etc.
- `value` may contained typed arrays, both using `ArrayBuffer`s

>    and `SharedArrayBuffer`s. and

Reviewer 4: ```suggestion
   and `SharedArrayBuffer`s, and
```

`value` may still contain `ArrayBuffer` instances that are not in
`transferList`; in that case, the underlying memory is copied rather than moved.

```js
const { MessageChannel } = require('worker_threads');
const { port1, port2 } = new MessageChannel();

port1.on('message', (message) => console.log(message));

const uint8Array = new Uint8Array([ 1, 2, 3, 4 ]);

> // This posts a copy of 'uint8Array':

Reviewer 4: ```suggestion
// This posts a copy of `uint8Array`:
```

port2.postMessage(uint8Array);
// This does not copy data, but renders `uint8Array` unusable:
port2.postMessage(uint8Array, [ uint8Array.buffer ]);

> // The memory for the sharedUint8Array will be accessible from both the

Reviewer 4: ```suggestion
// The memory for the `sharedUint8Array` will be accessible from both the
```

[`MessagePort`]: #worker_threads_class_messageport
[`SharedArrayBuffer`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer
[`Uint8Array`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array
[`WebAssembly.Module`]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/Module
[`Worker`]: #worker_threads_class_worker

> [`'close'` event]: #worker_threads_event_close

Reviewer 4: Should be first in the list ASCII-wise?
